esi/src/expression.rs

```diff
@@ -0,0 +1,1206 @@
+use fastly::http::Method;
+use fastly::Request;
+use regex::RegexBuilder;
+use std::fmt::Write;
+use std::iter::Peekable;
+use std::slice::Iter;
+use std::str::Chars;
+use std::{collections::HashMap, fmt::Display};
+
+use crate::{functions, ExecutionError, Result};
+
+pub fn try_evaluate_interpolated(
+    cur: &mut Peekable<Chars>,
+    ctx: &mut EvalContext,
+) -> Option<Value> {
+    evaluate_interpolated(cur, ctx)
+        .map_err(|e| {
+            println!("Error while evaluating interpolated expression: {e}");
```

Comment:
Should this throw an error?
Avoid `println`

Reply:
Removed the println and explained why we eat the error.